frontend/packages/topology/integration-tests/support/step-definitions/topology/chart-view.ts

```diff
@@ -0,0 +1,128 @@
+});
+
+Then('user will be redirected to Add page', () => {
+  cy.url().should('include', 'add');
+  app.waitForLoad();
+  cy.contains(pageTitle.Add).should('be.visible');
+});
+
+Given('user is at the Topology page', () => {
+  navigateTo(devNavigationMenu.Topology);
+});
+
+When('user clicks on {string} link in the empty topology page', (build: string) => {
+  cy.get(topologyPO.graph.startBuildingNewApp)
+    .should('be.visible')
+    .click();
+  cy.log(`Quick search bar can be seen with ${build} link`);
+});
+
+Then('user will be able to see Add to project search bar', () => {
+  cy.get(topologyPO.graph.quickSearch).should('be.visible');
+});
+
+Given('user has created a deployment workload named {string}', (componentName: string) => {
+  navigateTo(devNavigationMenu.Add);
+  createGitWorkload(
+    'https://github.com/sclorg/nodejs-ex.git',
+    componentName,
+    'Deployment',
+    'nodejs-ex-git-app',
+  );
+});
+
+Given('user has created deployment config workload {string}', (componentName: string) => {
+  navigateTo(devNavigationMenu.Add);
+  createGitWorkload(
+    'https://github.com/sclorg/nodejs-ex.git',
+    componentName,
+    'Deployment Config',
+    'nodejs-ex-git-app',
+  );
+});
+
+When('user navigates to Topology page', () => {
+  navigateTo(devNavigationMenu.Topology);
+});
+
+Then(
+  'user sees {string} and {string} workloads in topology chart area',
+  (workload1: string, workload2: string) => {
+    topologyHelper.verifyWorkloadInTopologyPage(workload1);
+    topologyHelper.verifyWorkloadInTopologyPage(workload2);
+  },
+);
+
+Given(
+  'user created {string} workload',
+  (resourceType: string, componentName: string = `${resourceType.toLowerCase()}-ex-git`) => {
+    navigateTo(devNavigationMenu.Add);
+    createGitWorkload(
+      'https://github.com/sclorg/django-ex.git',
+      componentName,
+      resourceType,
+      'django-ex-git-app',
+    );
+  },
+);
+
+When('user is at Topology page chart view', () => {
+  navigateTo(devNavigationMenu.Topology);
+});
+
+When('user clicks the filter by resource on top', () => {
+  cy.get(topologyPO.graph.filterByResource.filterByResourceDropDown)
+    .should('be.visible')
+    .click();
+});
+
+Then(
+  'user will see {string} and {string} options',
+  (resourceType1: string, resourceType2: string) => {
+    cy.byTestID(resourceType1).should('be.visible');
+    cy.byTestID(resourceType2).should('be.visible');
+  },
+);
+
+Then(
+  'user can see only the {string} workload when Deployment option is selected',
+  (workload: string) => {
+    cy.byTestID('Deployment').click();
+    cy.get(typeOfWorkload(workload)).should('be.visible');
+    cy.byTestID('Deployment').click();
+  },
+);
+
+Then(
+  'user can see only the {string} workload when Deployment-Config option is selected',
+  (workload: string) => {
+    cy.byTestID('DeploymentConfig').click();
+    cy.get(typeOfWorkload(workload)).should('be.visible');
+    cy.byTestID('DeploymentConfig').click();
```

Comment:
Is this back and forward clicks really required?

```suggestion
    cy.get(typeOfWorkload(workload)).should('be.visible');
```

Reply:
It's not required but I did to make it consistent with the previous then statement and if in future further test scenarios are added in the file, then we do not have to worry about the checks being disabled

Reply:
Sorry don't get the edge case where it is needed to click items twice.

Also, this is an `Then(...` condition. I would suggest not to click an items here anymore. If a click before that is needed this should be part of a `When` statement, or?

I would suggest to remove all clicks from

```tsx
Then(
  'user can see only the {string} workload when Deployment option is selected',
```

and

```tsx
Then(
  'user can see only the {string} workload when Deployment-Config option is selected',
```

Reply:
Updated